src/ark/phenotyping/pixie_preprocessing.py

```diff
@@ -308,9 +318,17 @@ def create_pixel_matrix(fovs, channels, base_dir, tiff_dir, seg_dir,
 
                     # drop the metadata columns and generate the 99.9% quantile values for the FOV
                     fov_full_pixel_data = pixel_mat_data.drop(columns=cols_to_drop)
-                    quant_dat[fov] = fov_full_pixel_data.replace(
-                        0, np.nan
-                    ).quantile(q=0.999, axis=0)
+                    quant_dat_fov = fov_full_pixel_data.replace(0, np.nan).quantile(
+                        q=channel_percentile_postnorm, axis=0).rename(fov)
+                    quant_dat_fov.index.name = "channel"
+
+                    # read in previously processed fov quantile values or initialize new df
+                    quant_dat_all = pd.read_csv(quantile_path, index_col="channel") \
```

Comment:
See comment above, we shouldn't need to keep loading in `quantile_path` on each iteration. Have it pre-loaded before the loop, then incrementally merge in new data and save while keeping it in memory.